Fix ppa podman (#1)

paulkitt opened the pull request:
The current used ppa is not recommended by the official podman install guide and not working with Ubuntu 20.04

paulkitt committed:
Fix ppa podman (#1)

* Add apt resource recomended bey podman dev

chasinglogic commented:
Do we need to add the upgrade command that is recommended by the official install docs?

paulkitt commented:
I think its better to follow the official docs because we trust their software anyway. When using another ppa we have to trust this other maintainer also. Adding ppas is always a risk because other packages can also be replaced. The ppa from projectatomic is also replacing the docker pkg for example.

Also the ppa is not supporting Ubuntu 20.04.

chasinglogic commented:
I'm not suggesting that we don't follow the official docs. I'm saying the official docs have an extra step that's not included in this PR.

The docs suggest running an apt upgrade before installing.

paulkitt commented:
Oh sorry for the missunderstanding :sweat_smile:. The `sudo apt-get update -qq` is executed already when adding the the pkg resource via the [ansible apt_repository module](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/apt_repository_module.html#parameter-update_cache).

chasinglogic commented:
Yes I'm aware. Per the official docs here are the steps they recommend a sudo apt **upgrade** as well as the apt update:

```
. /etc/os-release
echo "deb https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_${VERSION_ID}/ /" | sudo tee /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list
curl -L https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_${VERSION_ID}/Release.key | sudo apt-key add -
sudo apt-get update
sudo apt-get -y upgrade 
sudo apt-get -y install podman
```

You're PR covers these steps like so:

```
# Provided by add APT repository ansible module
echo "deb https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_${VERSION_ID}/ /" | sudo tee /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list
# Provided add_apt key ansible module
curl -L https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_${VERSION_ID}/Release.key | sudo apt-key add -
# Provided by add apt repository ansible module
sudo apt-get update
# This is the missing step we don't currently have or do.
sudo apt-get -y upgrade 
# Provided by apt package ansible module (already existing code)
sudo apt-get -y install podman
```

Note the step I've identified with a comment as missing the is the **upgrade** command. We will need to add that.

Commit:
Update system before installing

chasinglogic commented on tasks/ubuntu.yml:
```diff
@@ -1,10 +1,20 @@
   become_user: root
   apt_repository:
     repo: "{{ podman_apt_repo }}"
+    filename: podman
```
We should register a variable for this with

```
register: repository_added
```

chasinglogic commented on tasks/ubuntu.yml:
```diff
@@ -1,10 +1,20 @@
+
+- name: Ubuntu | Upgrade System before install
+  apt:
+    upgrade: dist
```
With the above comment we should then do:

```
when: repository_added.changed
```

Otherwise we'll do a system upgrade every time the user runs a playbook.

chasinglogic reviewed (changes requested):
This needs a README update as well letting users know that we'll be upgrade their systems per the Podman docs when we add the repository.

paulkitt committed:
Check if podman apt repo has changed before upgrading (#2)

Co-authored-by: Paul <kitt@dezim-institut.de>

Commit:
Update README: Info that install is like in the podman docs

chasinglogic commented:
@paulkitt This LGTM once the README has been updated I'll be happy to merge. Thanks!

paulkitt commented:
Thanks for your fast review yesterday! I accidentally read the dev section on the install page and then overlooked the `apt upgrade`. Added it now with the var tracking changes and added a reference to the podman docs in the readme file.

Commit:
Merge branch 'master' of github.com:coderat-collective/ansible-role-podman

chasinglogic commented on README.md:
```diff
@@ -7,7 +7,7 @@ Manage services and containers with podman
 -----
 
-By default this role will simply install podman and perform any other required
+By default this role will simply install podman as described in the [podman docs](https://podman.io/getting-started/installation.html) and perform any other required
 setup to make the installation functional on the supported platforms. 
```
```suggestion
By default this role will simply install podman as described in the 
[podman docs](https://podman.io/getting-started/installation.html) and perform any other required 
setup to make the installation functional on the supported platforms.

**Important Note:** This means that for Ubuntu / Debian systems it will perform an `apt upgrade` as per the docs after the repository is added. This upgrade will only be run if the repository is added or changed and will not execute on subsequent runs.
```

chasinglogic reviewed (changes requested):
One more minor change then we're good to go

Commit:
Add note to readme: Adding apt repo will perform upgrade

paulkitt commented:
Ok great! I added the note. Thanks for your great work with this role :)

chasinglogic commented:
I'll do a release this weekend to get this shipped

chasinglogic commented:
1.1.0 has been released with this change
